spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitUtils.java

```diff
@@ -369,4 +373,34 @@ public static int getMaxFrame(ConnectionFactory connectionFactory) {
 		return -1;
 	}
 
+	/**
+	 * Convert a String value to a {@link SaslConfig}.
+	 * Valid string values:
+	 * <ul>
+	 * <li>{@code DefaultSaslConfig.PLAIN}</li>
+	 * <li>{@code DefaultSaslConfig.EXTERNAL}</li>
+	 * <li>{@code JDKSaslConfig}</li>
+	 * <li>{@code CRDemoSaslConfig}</li>
+	 * </ul>
+	 * @param saslConfig the string value.
+	 * @param connectionFactory the connection factory to get the name, pw, host.
+	 * @return the saslConfig.
+	 */
+	public static SaslConfig stringToSaslConfig(String saslConfig,
+			com.rabbitmq.client.ConnectionFactory connectionFactory) {
+
+		switch (saslConfig) {
+		case "DefaultSaslConfig.PLAIN":
```

Comment:
String switch i not going to work with Java 6.
On the other hand  I don't understand why do you require a class name here?
Why just plain `plain` is not enough to request from end-user?

Reply:
Because `JDKSaslSupport` also supports `PLAIN`.